File: library/src/test/java/io/ashdavies/rx/rxtasks/SingleEmitterFailureListenerTest.java
package io.ashdavies.rx.rxtasks;

import io.reactivex.SingleEmitter;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import static org.mockito.BDDMockito.then;
import static org.mockito.Matchers.anyString;
import static org.mockito.Mockito.never;

@RunWith(MockitoJUnitRunner.class)
public class SingleEmitterFailureListenerTest {

  private SingleEmitterFailureListener<String> listener;

  @Mock SingleEmitter<String> emitter;
  @Mock Exception exception;

  @Before
  public void setUp() throws Exception {
    listener = new SingleEmitterFailureListener<>(emitter);
  }

  @Test
  public void shouldCallOnErrorWithException() throws Exception {
    listener.onFailure(exception);

    then(emitter).should(never()).onSuccess(anyString());
    then(emitter).should().onError(exception);
  }
}

File: library/src/test/kotlin/io/ashdavies/rx/rxtasks/SingleEmitterFailureListenerTest.kt
package io.ashdavies.rx.rxtasks

import com.nhaarman.mockito_kotlin.any
import com.nhaarman.mockito_kotlin.then
import io.reactivex.SingleEmitter
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.Mockito.never
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class SingleEmitterFailureListenerTest {

  private lateinit var listener: SingleEmitterFailureListener<String>

  @Mock private lateinit var emitter: SingleEmitter<String>
  @Mock private lateinit var exception: Exception

  @Before
  fun `set up`() {
    listener = SingleEmitterFailureListener(emitter)
  }

  @Test
  fun `should call on error with exception`() {
    listener.onFailure(exception)

    then(emitter).should(never()).onSuccess(any())
    then(emitter).should().onError(exception)
  }
}
